base/java-tools/src/com/netscape/cmstools/ca/CACertRequestApproveCLI.java

```diff
@@ -0,0 +1,87 @@
+package com.netscape.cmstools.ca;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.dogtagpki.cli.CommandCLI;
+
+import com.netscape.certsrv.ca.CACertClient;
+import com.netscape.certsrv.cert.CertRequestInfo;
+import com.netscape.certsrv.cert.CertReviewResponse;
+import com.netscape.certsrv.request.RequestId;
+import com.netscape.cmstools.cli.MainCLI;
+
+public class CACertRequestApproveCLI extends CommandCLI {
+
+    public static org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(CACertRequestApproveCLI.class);
+
+    CACertRequestCLI certRequestCLI;
+
+    public CACertRequestApproveCLI(CACertRequestCLI certRequestCLI) {
+        super("approve", "Approve certificate request", certRequestCLI);
+        this.certRequestCLI = certRequestCLI;
+    }
+
+    @Override
+    public void printHelp() {
+        formatter.printHelp(getFullName() + " <Request ID> [OPTIONS...]", options);
+    }
+
+    public void createOptions() {
+        Option option = new Option(null, "input-file", true, "Input file containing certificate request.");
+        option.setArgName("filename");
+        options.addOption(option);
+    }
+
+    @Override
+    public void execute(CommandLine cmd) throws Exception {
+
+        String[] cmdArgs = cmd.getArgs();
+
+        if (cmdArgs.length < 1) {
+            throw new Exception("Missing certificate request ID");
+        }
+
+        RequestId requestId;
+        try {
+            requestId = new RequestId(cmdArgs[0]);
+        } catch (NumberFormatException e) {
+            throw new Exception("Invalid certificate request ID: " + cmdArgs[0], e);
+        }
+
+        String filename = cmd.getOptionValue("input-file");
+
+        if (filename == null) {
+            throw new Exception("Missing input file");
```

Comment:
The deprecated method could approve a request directly **without** requiring a file:

````
pki <auth> ca-cert-request-review <Request ID> --action approve
````
So, the question is, should we make this `input-file` param mandatory?

Reply:
Yes, this is an old procedure which could approve a request without a review. As discussed on IRC, this is not really a recommended procedure.